### DISCUSSION TITLE
Return non-array JSON response

### DISCUSSION THREAD

**Discussion by brettks** [directus/directus#12809] [Archive]
I am attempting to interact with the directus API for simple tasks. I'm using the /items/:collection/?fields=xyz HTTP GET requests and successfully having JSON array returned. However, the program that is interpretting the response is unable to handle JSON array and needs to receive a JSON object only; the API request should only yield one item for each request. Is it possible to make the request such that directus outputs a single object as opposed to array in JSON format?

**Comment by rijkvanzanten** [comment]
The `/items/:collection` endpoint returns a list of items. You can use `/items/:collection/:pk` to fetch a single item by its key or configure the collection to be a Singleton to always have it act as a single item 👍🏻
